src/announcement/AnnouncementController.ts

```diff
@@ -0,0 +1,176 @@
+import { Announcement } from "entity/Announcement";
+import { Tag } from "entity/Tag";
+import { User } from "entity/User";
+import UserNotFoundException from "../exception/UserNotFoundException";
+import { Router, Response, Request, NextFunction } from "express";
+import validationMiddleware from "../middleware/validation-middleware";
+import userRepository from "../repository/UserRepository";
+import tagRepository from "../tag/TagRepository";
+import AddAnnouncementRequestDto from "./AddAnnouncementRequestDto";
+import announcementRepository from "./AnnnouncementRepository";
+import UpdateAnnouncementRequestDto from "./UpdateAnnouncementRequestDto";
+import AnnouncementNotFoundException from "./AnnouncementNotFoundException";
+import ChangeAnnouncementStatusRequestDto from "./ChangeAnnouncementStatusRequestDto";
+import querystring from "querystring";
+import { ParsedQs } from "qs";
+
+export default class AnnouncementController {
+
+    public path = "/announcement";
+    public router: Router = Router();
+
+    constructor() {
+        debugger;
+        this.initializeRoutes();
+    }
+
+    private initializeRoutes() {
+        this.router.post(`${this.path}`, validationMiddleware(AddAnnouncementRequestDto), this.addAnnouncement);
+        this.router.get(`${this.path}/:id`, this.getAnnouncementById);
+        this.router.get(`${this.path}s/`, this.getAllAnnouncements);
+        this.router.put(`${this.path}/:id`, validationMiddleware(UpdateAnnouncementRequestDto), this.updateAnnouncement);
+        this.router.put(`${this.path}/:id/status`, validationMiddleware(ChangeAnnouncementStatusRequestDto), this.updateAnnouncementStatus);
+    }
+
+    private async getAllAnnouncements(request: Request, response: Response, next: NextFunction) {
+        const anncouncementId : string = request.params.id;
+        const tags = request.query.tags + "";
```

Comment:
What is it? String convertion?
If so please use constructor 
```suggestion
        const tags = String(request.query.tags);
```